### PR TITLE
applet.sensor.hx711: modify to use port groups.

### PR DESCRIPTION
**Pull request by q3k**
Part of #599

### CODE REVIEW

**Review by whitequark — APPROVED** [review]
Thanks!

File: software/glasgow/applet/sensor/hx711/__init__.py [hunk 1]
```diff
@@ -16,40 +16,43 @@ class HX711Error(GlasgowAppletError):
...
         self.osc = Signal()
 
     def elaborate(self, platform):
         m = Module()
+        m.submodules.clk_buffer = clk_buffer = io.Buffer("o", self.ports.clk)
+        m.submodules.din_buffer = din_buffer = io.Buffer("i", self.ports.din)
         m.d.comb += [
-            self.pads.clk_t.oe.eq(1),
-            self.pads.clk_t.o.eq(self.clk),
+            clk_buffer.oe.eq(1),
```
Comment (whitequark) [inline software/glasgow/applet/sensor/hx711/__init__.py:30]: In `io.Buffer`, `oe` is high for `direction="o"`, so this line can be removed and only the `o.eq()` connection is necessary.

File: software/glasgow/applet/sensor/hx711/__init__.py [hunk 2]
```diff
@@ -16,40 +16,43 @@ class HX711Error(GlasgowAppletError):
...
-            FFSynchronizer(self.pads.din_t.i, self.din),
+            cdc.FFSynchronizer(din_buffer.i, self.din),
         ]
-        if hasattr(self.pads, "osc_t"):
+        if self.ports.osc is not None:
+            m.submodules.osc_buffer = osc_buffer = io.Buffer("o", self.ports.osc)
             m.d.comb += [
-                self.pads.osc_t.oe.eq(1),
-                self.pads.osc_t.o.eq(self.osc),
+                osc_buffer.oe.eq(1),
```
Comment (whitequark) [inline software/glasgow/applet/sensor/hx711/__init__.py:39]: Same here.

**Comment by q3k** [comment]
PTAL

File: software/glasgow/applet/sensor/hx711/__init__.py [hunk 3]
```diff
@@ -16,40 +16,41 @@ class HX711Error(GlasgowAppletError):
...
+        self.ports = ports
         self.clk = Signal()
         self.din = Signal()
         self.osc = Signal()
 
     def elaborate(self, platform):
         m = Module()
+        m.submodules.clk_buffer = clk_buffer = io.Buffer("o", self.ports.clk)
+        m.submodules.din_buffer = din_buffer = io.Buffer("i", self.ports.din)
         m.d.comb += [
```
Comment (whitequark) [inline software/glasgow/applet/sensor/hx711/__init__.py:29]: I think the code gets a bit nicer if you compactify it to transform

```
m.d.comb += [
    xxx
]
```

to the equivalent

```
m.d.comb += xxx
```

Comment (q3k) [reply software/glasgow/applet/sensor/hx711/__init__.py]: PTAL

**Review by whitequark — COMMENTED** [review]
Perfect! Feel free to go ahead and do the same to the rest, this is more than good enough.

**Pull request merged**